packages/protvista-track/src/FeatureShape.js

```diff
@@ -4,233 +4,620 @@
+    return this._getPolygon(6);
+  }
+
+  _pentagon() {
+    return this._getPolygon(5);
+  }
+
+  _circle() {
+    const m = this._ftWidth / 2;
+    const r = Math.sqrt(symbolSize / Math.PI);
+    const shape =
+      "M" +
+      m +
+      ",0" +
+      "A" +
+      r +
+      "," +
+      r +
+      " 0 1,1 " +
+      m +
+      "," +
+      symbolSize +
+      "A" +
+      r +
+      "," +
+      r +
+      " 0 1,1 " +
+      m +
+      ",0";
+    return this._ftLength !== 1
+      ? shape + this._getMiddleLine(symbolSize / 2, this._ftWidth) + "Z"
+      : shape + "Z";
+  }
+
+  _arrow() {
+    const step = symbolSize / 10;
+    const m = this._ftWidth / 2;
+    const shape =
+      "M" +
+      m +
+      ",0" +
+      "L" +
+      (-step + m) +
+      ",0" +
+      "L" +
+      (-5 * step + m) +
+      "," +
+      4 * step +
+      "L" +
+      (-step + m) +
+      "," +
+      this._ftHeight +
+      "L" +
+      m +
+      "," +
+      this._ftHeight +
+      "L" +
+      (4 * step + m) +
+      "," +
+      4 * step;
+    return this._ftLength !== 1
+      ? shape +
+          "L" +
+          m +
+          ",0" +
+          this._getMiddleLine(symbolSize / 2, this._ftWidth) +
+          "Z"
+      : shape + "Z";
+  }
+
+  _doubleBar() {
+    const m = this._ftWidth / 2;
+    const centerx = symbolSize / 2;
+    const shape =
+      "M" +
+      m +
+      ",0" +
+      "L" +
+      (-centerx + m) +
+      "," +
+      symbolSize +
+      "L" +
+      m +
+      "," +
+      symbolSize +
+      "L" +
+      (centerx + m) +
+      ",0";
+    return this._ftLength !== 1
+      ? shape +
+          "L" +
+          m +
+          ",0" +
+          this._getMiddleLine(symbolSize / 2, this._ftWidth) +
+          "Z"
+      : shape + "Z";
+  }
+
+  _getBrokenEnd() {
+    const qh = this._ftHeight / 4.0;
+    return (
+      "L" +
+      (this._ftWidth - qh) +
+      "," +
+      qh +
+      "L" +
+      this._ftWidth +
+      "," +
+      2 * qh +
+      "L" +
+      (this._ftWidth - qh) +
+      "," +
+      3 * qh +
+      "L" +
+      this._ftWidth +
+      "," +
+      this._ftHeight
+    );
+  }
+  _getBrokenStart() {
+    const qh = this._ftHeight / 4.0;
+    return "L" + qh + "," + 3 * qh + "L0," + 2 * qh + "L" + qh + "," + qh;
+  }
+  _discontinuosStart() {
+    return (
+      "M0,0" +
+      "L" +
+      this._ftWidth +
+      ",0" +
+      "L" +
+      this._ftWidth +
+      "," +
+      this._ftHeight +
+      "L0," +
+      this._ftHeight +
+      this._getBrokenStart() +
+      "Z"
+    );
+  }
+  _discontinuos() {
+    return (
+      "M0,0" +
+      "L" +
+      this._ftWidth +
+      ",0" +
+      this._getBrokenEnd() +
+      "L0," +
+      this._ftHeight +
+      this._getBrokenStart() +
+      "Z"
+    );
+  }
+  _discontinuosEnd() {
+    return (
+      "M0,0" +
+      "L" +
+      this._ftWidth +
+      ",0" +
+      this._getBrokenEnd() +
+      "L0," +
+      this._ftHeight +
+      "Z"
+    );
+  }
+  _helix() {
```

Comment:
Unfortunately prettier has made it hard to identify that the addition to the file is the last 2 shapes.

The created path is also harder to read, maybe we should rewrite this file using string literals \`\`??